terraform/azure/modules/monitoring/main.tf

```diff
@@ -64,57 +64,59 @@ module "os_image_reference" {
   os_image_srv_uri = var.monitoring_uri != ""
 }
 
-resource "azurerm_virtual_machine" "monitoring" {
-  name                             = var.name
-  count                            = var.monitoring_enabled == true ? 1 : 0
-  location                         = var.az_region
-  resource_group_name              = var.resource_group_name
-  network_interface_ids            = [azurerm_network_interface.monitoring.0.id]
-  vm_size                          = var.vm_size
-  delete_os_disk_on_termination    = true
-  delete_data_disks_on_termination = true
-
-  storage_os_disk {
-    name              = "disk-monitoring-Os"
-    caching           = "ReadWrite"
-    create_option     = "FromImage"
-    managed_disk_type = "Premium_LRS"
-  }
+resource "azurerm_managed_disk" "monitoring_data" {
+  count                = var.monitoring_enabled == true ? 1 : 0
+  name                 = "disk-monitoring-Data01"
```

Comment:
In VM, the name is controllable by a variable before

https://github.com/SUSE/qe-sap-deployment/blob/c35d2805b4697d78916e4d6c6d061ea3671af345/terraform/azure/modules/monitoring/variables.tf#L20

Does not make sense to attach the name here too? Maybe not?

Reply:
This is the data disk, and it had a set name from before - I just kept it as is:

```
storage_data_disk {
    name              = "disk-monitoring-Data01"
    caching           = "ReadWrite"
    create_option     = "Empty"
    disk_size_gb      = "10"
    lun               = "0"
    managed_disk_type = "Standard_LRS"
```